File: lib/domain/cultivation_plan/dtos/optimization_plan_read_plan_core_snapshot.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      class OptimizationPlanReadPlanCoreSnapshot
        attr_reader :plan_id,
                    :plan_type_private,
                    :calculated_planning_start_date,
                    :calculated_planning_end_date,
                    :prediction_target_end_date,
                    :predicted_weather_data,
                    :total_area,
                    :weather_location_present

        def initialize(
          plan_id:,
          plan_type_private:,
          calculated_planning_start_date:,
          calculated_planning_end_date:,
          prediction_target_end_date:,
          predicted_weather_data:,
          total_area:,
          weather_location_present:
        )
          @plan_id = plan_id
          @plan_type_private = plan_type_private
          @calculated_planning_start_date = calculated_planning_start_date
          @calculated_planning_end_date = calculated_planning_end_date
          @prediction_target_end_date = prediction_target_end_date
          @predicted_weather_data = predicted_weather_data
          @total_area = total_area
          @weather_location_present = weather_location_present
          freeze
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/optimization_plan_read_plan_core_snapshot.rs
//! Ruby: `Domain::CultivationPlan::Dtos::OptimizationPlanReadPlanCoreSnapshot`

use time::Date;

use crate::weather_data::dtos::PredictedWeatherMetadata;

/// Ruby: `Domain::CultivationPlan::Dtos::OptimizationPlanReadPlanCoreSnapshot`
#[derive(Debug, Clone)]
pub struct OptimizationPlanReadPlanCoreSnapshot {
    pub plan_id: i64,
    pub plan_type_private: bool,
    pub calculated_planning_start_date: Option<Date>,
    pub calculated_planning_end_date: Option<Date>,
    pub prediction_target_end_date: Option<Date>,
    pub plan_metadata: Option<PredictedWeatherMetadata>,
    pub total_area: Option<f64>,
    pub weather_location_present: bool,
}
